packages/web-app/src/modules/notifications/models/NotificationMessage.tsx

```diff
@@ -52,3 +54,58 @@ export interface NotificationMessage {
    */
   onClick?: () => void
 }
+
+/** A resource that represents an action that acknowledges a notification. */
+export interface AcknowledgeNotificationAction {
+  /** The title. */
+  title: string;
+}
+/** A resource that represents an action that dismisses a notification. */
+export interface DismissNotificationAction {
+  /** The title. */
+  title: string;
+}
+/** A resource that represents an action that opens a link in the default browser. */
+export interface OpenLinkNotificationAction {
+  /** The title. */
+  title: string;
+  /** The link. */
+  link: string;
+}
+
+/** A resource that represents an in-app notification action. */
+export interface NovuNotificationAction {
+  action?: {
+      $case: "acknowledge";
+      acknowledge: AcknowledgeNotificationAction;
+  } | {
+    $case: "dismiss";
+    dismiss: DismissNotificationAction;
+  } | {
+      $case: "openLink";
+      openLink: OpenLinkNotificationAction;
+  }
+}
+
+export interface NovuNotification {
```

Comment:
Seems the naming is misleading here: this interface describes the application needs, but not describe the content of novu notification
```suggestion
export interface Notification {
```